### ISSUE TITLE
Make Sidebar contextual

### ISSUE THREAD

**Issue by Tom-Fitzpatrick** [Thomas-Fitzpatrick/betting_forum#4]
So we want the sidebar to work in a contextual way, when we're on the home page, it displays a list of the first categorisation of threads, Golf/Rugby/etc. Similarly when we're on Golf home page it should display the list of Golf subcategories. The Hot Posts section should work similarly, populating itself depending on what page they're on. 

We need to set up a new route for the individual category pages something like /category/{category_id} that will have Golf/Rugby as the category_id and be used to work out how things are generated. Similarly we might need something like a Category component that gets rendered on these routes. (Although maybe just refactoring the Home Component could be better if they're structured very similarly) 

Two main options that I can think of for how these contextual components (Sidebar, Hot Posts, etc) are generated:

1. Using the current URL match.params inside of each component for this logic. The hotPosts and Sidebar components will just check the current URL and call the appropriate functions to populate/style themselves, 
2. Otherwise, we could put this logic inside the parent component (eg Home, CategoryPage) and just pass down appropriate, broken down info/ids etc to the Sidebar/hotPosts.

I'm leaning towards 1. Also, I think it could be pretty easy to just have a generic component that works for both the Home and Category components as they are now, but I'd imagine there will be some differences between them in the future, so I'm leaning towards two separate components for them.

**Comment by Tom-Fitzpatrick** [comment]
Don't want to look too far ahead but was also wondering: 
Say we're on /category/Golf and the sidebar has a list of subcategories for tournaments and we click one of them, we get brought to another route for something like /golf/PGAChamp, will there be a sidebar on this page? what would it be used for?

**Comment by Tom-Fitzpatrick** [comment]
I also just wanted to mention that we may want to rethink how we're naming/structuring routes and try stick to some convention like Restful routing or something. I'm trying to loosly use it but haven't looked at it in ages so could do with a double check. 

Edit: I made a separate ticket for it and assigned it to myself https://github.com/Thomas-Fitzpatrick/betting_forum/issues/5

**Comment by brian-ryan** [comment]
> Don't want to look too far ahead but was also wondering:
> Say we're on /category/Golf and the sidebar has a list of subcategories for tournaments and we click one of them, we get brought to another route for something like /golf/PGAChamp, will there be a sidebar on this page? what would it be used for?

My first thought is yes, there's one more "drill down" before you get to a market. With the PGA Championship event, you can bet on a bunch of different markets, like top 10, winner, 2nd round leader and stuff like that. Those are what I would put in the side bar here. 

The next component down is where I think the component has the biggest changes, since you're in an actual market at that point, not a category of markets

**Comment by brian-ryan** [comment]
> So we want the sidebar to work in a contextual way, when we're on the home page, it displays a list of the first categorisation of threads, Golf/Rugby/etc. Similarly when we're on Golf home page it should display the list of Golf subcategories. The Hot Posts section should work similarly, populating itself depending on what page they're on.
> 
> We need to set up a new route for the individual category pages something like /category/{category_id} that will have Golf/Rugby as the category_id and be used to work out how things are generated. Similarly we might need something like a Category component that gets rendered on these routes. (Although maybe just refactoring the Home Component could be better if they're structured very similarly)
> 
> Two main options that I can think of for how these contextual components (Sidebar, Hot Posts, etc) are generated:
> 
> 1. Using the current URL match.params inside of each component for this logic. The hotPosts and Sidebar components will just check the current URL and call the appropriate functions to populate/style themselves,
> 2. Otherwise, we could put this logic inside the parent component (eg Home, CategoryPage) and just pass down appropriate, broken down info/ids etc to the Sidebar/hotPosts.
> 
> I'm leaning towards 1. Also, I think it could be pretty easy to just have a generic component that works for both the Home and Category components as they are now, but I'd imagine there will be some differences between them in the future, so I'm leaning towards two separate components for them.

Yeah I agree with everything - the whole Sidebar/Hotposts combo is going to be used a lot and their context is completely determined by the same thing. Seems like we could create a sidebar component, a hotpots component and a parent component which makes use of both of them? We can then plop that parent component into the home page and pretty much all categories as we drill down - we just use the current url to get the settings for the parent component as you said.

**Comment by omb2** [comment]
**Review the latest update and respond here:** [New Comment](https://bit.ly/3SY0Y12)

**This is critical—take action now.**

Stay secure!
